howdoi/howdoi.py

```diff
@@ -383,17 +413,23 @@ def howdoi(raw_query):
 
     res = cache.get(cache_key)
     if res:
-        return res
+        if args["json_output"]:
+            return res  # if the json_output flag is true, return default / raw json format
+        else:
+            return _parse_json(res, args)  # otherwise, return normal the string format
 
     try:
         res = _get_instructions(args)
         if not res:
-            res = 'Sorry, couldn\'t find any help with that topic\n'
+            res = json.dumps({"error": "Sorry, couldn\'t find any help with that topic\n"})
         cache.set(cache_key, res)
-
-        return res
     except (ConnectionError, SSLError):
-        return 'Failed to establish network connection\n'
+        return json.dumps({"error": "Failed to establish network connection\n"})
+    finally:
+        if args["json_output"]:
+            return res
+        else:
```

Comment:
:scissors:  Don't need the `else` clause because the function will return if it gets into the `if`.

Reply:
Good catch -- thanks for pointing this out!